engine/storage/image/src/main/java/org/apache/cloudstack/storage/image/store/TemplateObject.java

```diff
@@ -93,10 +95,15 @@ public static TemplateObject getTemplate(VMTemplateVO vo, DataStore store, Strin
     }
 
     public void setSize(Long size) {
-        imageVO.setSize(size);
+        getImage().setSize(size);
     }
 
     public VMTemplateVO getImage() {
+        if (imageVO == null) {
```

Comment:
@DaanHoogland if `imageVO` shouldn't be null for all these get calls, better get that initialized during the object creation itself (in constructor), instead of throwing errors on get calls.

Reply:
That is not how it is used in the code I'm afraid. The constructor is called and next configure is called with the undelying template object. I can move the exception to the configure call, if we think that makes ore sense.. We'd still have to guard during the get in case configure was never called.

Reply:
@sureshanaparti , I changed the class to have a private constructor, making sure the configure is always called and the input image is checked. (needs extensive testing ;)

Reply:
@suresh, this tactic fails as some `TemplateObject`s are created from spring contexts. I think the null check needs to go back into the `getImage()` method.